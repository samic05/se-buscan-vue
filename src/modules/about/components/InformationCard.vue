<template>
  <div class="card">
    <img :src="information.image" class="avatar" alt="Avatar" />
    <div class="container">
      <h4>
        <b>{{ information.title }}</b>
      </h4>
      <p>
        {{ information.content }}
      </p>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    information: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {};
  },
};
</script>

<style scoped>
@import url("https://unpkg.com/css.gg@2.0.0/icons/css/arrow-right.css");
.card {
  border-radius: 20px;
  padding: 10px;
  margin: 20px;
  width: 400px;
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
}
.avatar {
  height: 130px;
  width: 200px;
  display: block;
  margin-left: auto;
  margin-right: auto;
  width: 50%;
}

/* Add some padding inside the card container */
.container {
  padding: 2px 16px;
  font-family: "Poppins", sans-serif;
}

.container h4 {
  font-weight: 600;
  font-size: 20px;
}
.container p {
  font-weight: 400;
  font-size: 14px;
  color: #525252;
}

.read {
  font-weight: 500;
  font-size: 14px;
  display: flex;
  justify-items: center;
  justify-content: flex-end;
  bottom: 25px;
  right: 30px;
}
</style>
