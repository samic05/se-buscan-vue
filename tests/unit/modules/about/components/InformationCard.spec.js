import InformationCard from '@/modules/about/components/InformationCard'
import { shallowMount } from "@vue/test-utils"

describe('InformationCardComponent',() => {

    test('should match with snapshot', () => {
        const informationCardComponentWrapped =  shallowMount(InformationCard,{
            props:{
                information: {
                    image: require("@/assets/about-icons/social-network.svg"),
                    title: "Red Social",
                    content:
                      "Para la comunidad de desaparecidos en el mundo. Sus familiares o amigos podrán vincularse todas las entidades involucradas en la problemática de desaparición. SE BUSCAN será el nodo mundial de información dedesaparecidos.",
                  }
            }
        });
 
     expect(informationCardComponentWrapped.html()).toMatchSnapshot()
 
     })
})