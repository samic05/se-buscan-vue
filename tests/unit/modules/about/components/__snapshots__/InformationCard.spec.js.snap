// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InformationCardComponent should match with snapshot 1`] = `
<div class="card"><img src="" class="avatar" alt="Avatar">
  <div class="container">
    <h4><b>Red Social</b></h4>
    <p>Para la comunidad de desaparecidos en el mundo. Sus familiares o amigos podrán vincularse todas las entidades involucradas en la problemática de desaparición. SE BUSCAN será el nodo mundial de información dedesaparecidos.</p>
  </div>
</div>
`;
